streaming/src/main/scala/org/apache/spark/streaming/dstream/TransformedDStream.scala

```diff
@@ -38,6 +39,12 @@ class TransformedDStream[U: ClassTag] (
 
   override def compute(validTime: Time): Option[RDD[U]] = {
     val parentRDDs = parents.map(_.getOrCompute(validTime).orNull).toSeq
-    Some(transformFunc(parentRDDs, validTime))
+    val transformedRDD = transformFunc(parentRDDs, validTime)
+    if (transformedRDD == null) {
```

Comment:
I don't have a strong opinion on this, but this could be done with `require`. Generates a different exception -- `IllegalArgumentException`, but a lower-level generic exception seems pretty reasonable.

Can I suggest tightening the text to something like "Transform function may not return null. Return RDD.empty to return no elements as the result of the transformation."